src/app/static/src/app/store/modelCalibrate/actions.ts

```diff
@@ -159,11 +156,28 @@ export const actions: ActionTree<ModelCalibrateState, RootState> & ModelCalibrat
                 } as CalibrateResultWithType
                 commit({type: ModelCalibrateMutation.CalibrateResultFetched, payload: payload});
             }
-            commit(`modelOutput/${ModelOutputMutation.SetTabLoading}`, {payload:{tab, loading: false}}, {root: true});
+            commit(`modelOutput/${ModelOutputMutation.RemoveIndicatorBeingFetched}`, {payload: indicatorId}, {root: true});
         }
     }
 };
 
+export const getResultDataForNIndicators = async (
+    context: ActionContext<ModelCalibrateState, RootState>,
+    numOfIndicators: number,
+    indicators: BarchartIndicator[]
+) => {
+    const {dispatch} = context;
+    const indicatorIds: string[] = [];
+    indicators.forEach((indicator, index) => {
+        if (index >= numOfIndicators) return;
+        indicatorIds.push(indicator.indicator);
+    });
+    const promisesArray = indicatorIds.map(indicatorId => {
+        return dispatch("getResultData", indicatorId);
+    });
+    await Promise.all(promisesArray);
```

Comment:
so if i were to stick an await in front of the caller to the function, then it would wait until all the results have been fetched basically, i dont have to do it this way, i could equally just fire off 5 dispatches and not put this await line at all, that would also work i think but its often nice to be able to let people actually wait stuff out if they put an await keyword in front of the caller because if that await line wasnt there and i put an await keyword in front of the caller i dont think the function would actually wait until everything is called

take this with a pinch of salt though, i half understand what it does under the hood but would also like to learn more about it properly